app/uk/gov/hmrc/teamsandrepositories/services/JenkinsService.scala

```diff
@@ -47,15 +59,39 @@ class JenkinsService @Inject()(repo: BuildJobRepo, jenkinsConnector: JenkinsConn
       persist <- repo.update(buildJobs)
     } yield persist
 
-  def triggerBuildJob(serviceName:String, url: String,
-                      timestamp: Instant)(implicit ec: ExecutionContext): Future[Unit] = {
+  def triggerBuildJob(serviceName: String, url: String,
+                      timestamp: Instant)(implicit ec: ExecutionContext): Future[JenkinsBuildData] = {
     for {
       latestBuild <- jenkinsConnector.getLastBuildTime(url)
-      _ <- if (latestBuild.timestamp.equals(timestamp)) {
-          logger.info(s"Triggering build for ${serviceName}")
-          jenkinsConnector.triggerBuildJob(url)
-        }
-        else Future.successful[Unit](())
-    } yield ()
+      location <- {
+        logger.info(s"Triggering build for $serviceName")
+        jenkinsConnector.triggerBuildJob(url)
+      } if latestBuild.timestamp.equals(timestamp)
+      queueUrl = s"${location.replace("http:", "https:")}api/json"
+      queue <- getQueue(queueUrl)
+      build <- getBuild(queue.executable.get.url)
+    } yield build
   }
+
+  private def getQueue(queueUrl: String)(implicit ec: ExecutionContext): Future[JenkinsQueueData] = {
+    Source.repeat(42)
```

Comment:
Move `42` to config too?

Reply:
This is just a random value to create an infinite stream and has no meaning for our code other than you need a value.

Reply:
pick a more neutral value - like `()`?